packages/db/src/translate/file.ts

```diff
@@ -5,12 +5,12 @@ import { PostgresFile } from '../types/file';
 export const translateApiFiletoPostgresFile = (
   file: ApiFile
 ): Omit<PostgresFile, 'granule_cumulus_id'> => ({
-  bucket: file.bucket,
+  bucket: file.bucket || '',
```

Comment:
What is the cause for this? I guess a `bucket` and `key` that are empty strings are technically valid, but it's going to make looking up records uniquely quite difficult.

I guess more precisely: why are we allowing empty strings based on what we found from the investigation of the PODAAC data?

Reply:
Yup! Apologies, I wanted to check that this actually did what I thought it would before commenting. Wasn't trying to slip this through 😄 

The intention in this PR is to make file.bucket and file.key required fields in the Postgres schema because the PODAAC data indicates these files will likely always be present.  We also need to update the `PostgresFile` TypeScript type definition to make these fields required.

The trick here is that the Dynamo `ApiFile` TypeScript type definition does NOT have bucket/key as required fields. I think that's correct; those fields are not required by Dynamo so making them required on the `ApiFile` doesn't feel right. It seems like the TypeScript definition should match the table schema. What that means here is that we're trying to assign the Dynamo `string or undefined` type to the Postgres `string` type. Which isn't possible. This change defaults to an empty string if bucket or file is undefined to make the TypeScript compiler happy.

We could talk about making the bucket/key properties required on `ApiFile`. That's what I initially did that caused the integration test failures. In retrospect I don't think that's what we want regardless of test failures but I could be wrong.

This could be worth a team discussion.

Reply:
Put more succinctly, the question here is:

What should we do if we find a file without a bucket or key during Dynamo to Postgres migration?

Based on the PODAAC data, this may never be the case (it isn't in PODAAC's).

Reply:
Discussed offline: we will throw an error here if `bucket` or `key` is undefined. Will reply with commit # once all integration tests are verified.

Reply:
Well, it's several commits 😄. `translateApiFiletoPostgresFile` now throws an error and is handled in write-granules.js.

Reply:
Going to resolve this discussion as it's ~4 refactors out of date and likely to cause confusion.